src/mscorlib/shared/System/BitConverter.cs

```diff
@@ -21,6 +21,20 @@ public static class BitConverter
 #else
         public static readonly bool IsLittleEndian = true;
 #endif
+        private static T ReadUnaligned<T>(ReadOnlySpan<byte> value) where T : struct
+        {
+            if (value.Length < Unsafe.SizeOf<T>())
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.value);
+            return Unsafe.ReadUnaligned<T>(ref value.DangerousGetPinnableReference());
+        }
+
+        private static bool WriteUnaligned<T>(T value, Span<byte> bytes) where T : struct
+        {
+            if (bytes.Length < Unsafe.SizeOf<T>())
```

Comment:
It would be better to inline the implementation instead of using generics, like it is done for the existing array version. Generics have overhead.